src/main/java/org/commonwl/view/git/GitService.java

```diff
@@ -72,53 +73,63 @@ public GitService(@Value("${gitStorage}") Path gitStorage,
     public Git getRepository(GitDetails gitDetails, boolean reuseDir)
             throws GitAPIException {
         Git repo = null;
-        try {
-            if (reuseDir) {
-                // Base dir from configuration, name from hash of repository URL
-                File baseDir = new File(gitStorage.toString());
-                String baseName = DigestUtils.sha1Hex(GitDetails.normaliseUrl(gitDetails.getRepoUrl()));
-
-                // Check if folder already exists
-                File repoDir = new File(baseDir, baseName);
-                if (repoDir.exists() && repoDir.isDirectory()) {
-                    repo = Git.open(repoDir);
-                    repo.fetch().call();
-                } else {
-                    // Create a folder and clone repository into it
-                    if (repoDir.mkdir()) {
-                        repo = Git.cloneRepository()
-                                .setCloneSubmodules(cloneSubmodules)
-                                .setURI(gitDetails.getRepoUrl())
-                                .setDirectory(repoDir)
-                                .setCloneAllBranches(true)
-                                .call();
+        while (repo == null) {
+            try {
+                if (reuseDir) {
+                    // Base dir from configuration, name from hash of repository URL
+                    File baseDir = new File(gitStorage.toString());
+                    String baseName = DigestUtils.sha1Hex(GitDetails.normaliseUrl(gitDetails.getRepoUrl()));
+
+                    // Check if folder already exists
+                    File repoDir = new File(baseDir, baseName);
+                    if (repoDir.exists() && repoDir.isDirectory()) {
+                        repo = Git.open(repoDir);
+                        repo.fetch().call();
+                    } else {
+                        // Create a folder and clone repository into it
+                        if (repoDir.mkdir()) {
```

Comment:
Would prefer using `Path` and `Files.createDirectory()` to throw exception if it breaks rather than no-op..